pilot/pkg/networking/core/v1alpha3/gateway_test.go

```diff
@@ -2263,3 +2263,115 @@ func TestBuildNameToServiceMapForHttpRoutes(t *testing.T) {
 		t.Errorf("The value of hostname %s mapping must be exist and it should be nil.", bazHostName)
 	}
 }
+
+func TestBuildGatewayListenersFilters(t *testing.T) {
+	cases := []struct {
+		name                   string
+		node                   *pilot_model.Proxy
+		gateways               []config.Config
+		virtualServices        []config.Config
+		expectedHTTPFilters    []string
+		expectedNetworkFilters []string
+	}{
+		{
+			"http",
+			&pilot_model.Proxy{Type: pilot_model.Router},
+			[]config.Config{
```

Comment:
nit: use key names for each struct value. It is hard to tell what is what since we have multiple elements of the same type. Goland can auto add them